realm/realm-library/src/main/java/io/realm/RealmResults.java

```diff
@@ -329,6 +329,8 @@ private long getColumnIndexForSort(String fieldName) {
 
     /**
      * {@inheritDoc}
+     *
+     * @return a new sorted {@link RealmResults}. The original {@link RealmResults} stays unchanged.
```

Comment:
I would move this to `OrderedRealmCollection` instead and then perhaps rephrase to `@return a new sorted {@link RealmResults}. The original collection stays unchanged.`